File: 2022/day3.py
def puzzle_2():
    with open("./2022/day3_input.txt", "r") as file:
        input = file.read().splitlines()
    total = sum(
        list(
            map(
                lambda x: x - 96 if x >= 97 and x <= 123 else x - 38,
                [
                    ord(
                        set(input[i])
                        .intersection(set(input[i + 1]), set(input[i + 2]))
                        .pop()
                    )
                    for i in range(0, len(input), 3)
                ],
            )
        )
    )
    print(total)


puzzle_2()

File: 2022/day3.go
package main

import (
	"fmt"
	"os"
	"strings"
)

func puzzle1(input string) {
	rucksacks := strings.Split(input, "\n")
	var total_priority int
	for _, rucksack := range rucksacks {
		compartment1 := rucksack[:len(rucksack)/2]
		compartment2 := rucksack[len(rucksack)/2:]
		var shared_item int
		for _, item := range compartment1 {
			if strings.Contains(compartment2, string(item)) {
				shared_item = int(item)
				if shared_item >= 97 && shared_item <= 123 {
					total_priority += shared_item - 96
				} else {
					total_priority += shared_item - 38
				}
				break
			}
		}
	}
	fmt.Println(total_priority)
}

func main() {
	raw_input, _ := os.ReadFile("./day3_input.txt")
	input := string(raw_input)
	puzzle1(input)
}
